packages/runtime/container-runtime/src/summaryStorageCompressionHooks.ts

```diff
@@ -0,0 +1,199 @@
+            const decompressed: Uint8Array = typeof summaryBlob.content === "string" ?
+                new TextEncoder().encode(summaryBlob.content) : summaryBlob.content;
+            if (this._minSizeToCompress !== undefined && decompressed.length < this._minSizeToCompress) {
+                return input;
+            }
+            const compressed: ArrayBufferLike = this.encodeBlob(decompressed);
+            let newSummaryBlob;
+            if (this._isUseB64OnCompressed !== undefined && this._isUseB64OnCompressed) {
+                // TODO: This step is now needed, it looks like the function summaryTreeUploadManager#writeSummaryBlob
+                // fails on assertion at 2 different generations of the hash which do not lead to
+                // the same result if the ISummaryBlob.content is in the form of ArrayBufferLike
+                const compressedString = Uint8ArrayToString(IsoBuffer.from(compressed), "base64");
+                const compressedEncoded = new TextEncoder().encode(compressedString);
+                newSummaryBlob = { type: SummaryType.Blob, content: compressedEncoded };
+            } else {
+            // This line will replace the 3 lines above when the bug is fixed.
+            // const newSummaryBlob: ISummaryBlob = { type: SummaryType.Blob, content: IsoBuffer.from(compressed)};
+                newSummaryBlob = { type: SummaryType.Blob, content: IsoBuffer.from(compressed) };
+            }
+            return newSummaryBlob;
+        } else {
+            return input;
+        }
+    };
+    constructor(private readonly _algorithm: SummaryCompressionAlgorithms | undefined,
+        private readonly _minSizeToCompress: number | undefined,
+        private readonly _isUseB64OnCompressed: boolean | undefined) {
+            if (this._minSizeToCompress === undefined) {
+                this._minSizeToCompress = defaultMinSizeToCompress;
+            }
+        }
+    public onPreCreateBlob(file: ArrayBufferLike): ArrayBufferLike {
+        return this.encodeBlob(file);
+    }
+    public onPostReadBlob(file: ArrayBufferLike): ArrayBufferLike {
+        return this.decodeBlob(file);
+    }
+
+    /**
+     * All paths of ISummaryTree which lead to ISummaryBlob objects are iterated.
+     * At each ISummaryBlob, content is mapped to binary array and compressed.
+     * The header is then addded which shows, which algorithm was used for encryption.
+     * The result binary is base64 encoded due to the issue with summaryTreeUploadManager#writeSummaryBlob
+     * hash assertion
+     * New ISummaryBlob is created with the new string content obtained in the above step and the
+     * old blob is replaced by this new blob in the ISummaryTree.
+     */
+    public onPreUploadSummaryWithContext(summary: ISummaryTree, context: ISummaryContext):
+    { prepSummary: ISummaryTree; prepContext: ISummaryContext; } {
+        console.log("Using Summary Compression : UploadSummaryWithContext ");
```

Comment:
Nit: Should remove, convert to telemetry, or use DebugLogger.

Reply:
Fixed in 023009c70d848ebeae729f7c73de6432185435e2